pandas/tests/series/test_query_eval.py

```diff
@@ -0,0 +1,400 @@
+            "  &^ :!€$?(} >    <++*''  "))
+        run_test(series, "1 < `  &^ :!€$?(} >    <++*''  `")
+
+    def test_failing_character_outside_range(self):
+        series = Series(np.random.randn(10), index=Index(list(range(10)), name="☺"))
+        with pytest.raises(SyntaxError):
+            series.query("`☺` > 4")
+
+    def test_failing_hashtag(self):
+        series = Series(np.random.randn(10), index=Index(list(range(10)), name="foo#bar"
+            ))
+        with pytest.raises(SyntaxError):
+            series.query("`foo#bar` > 4")
+
+
+class TestSeriesQueryWithMultiIndex:
+    def setup_method(self, method):
+        multiIndex = MultiIndex.from_arrays([["a"] * 5 + ["b"] * 5, [str(x) for x in
+            range(10)]], names=["alpha", "num"])
+        self.series = Series(np.random.randn(10), index=multiIndex)
+
+    def teardown_method(self, method):
+        del self.series
+
+    # test against level 0
+    def test_equality(self):
+        for op in ["==", "!="]:
+            run_test(self.series, "alpha " + op + " 'b'")
+            run_test(self.series, "'b' " + op + " alpha")
+
+    def test_list_equality(self):
+        for op in ["==", "!="]:
+            run_test(self.series, "alpha " + op + " ['b']")
+            run_test(self.series, "['b'] " + op + " alpha")
+
+    def test_in_operators(self):
+        for op in ["in", "not in"]:
+            run_test(self.series, "['b'] " + op + " alpha")
+            run_test(self.series, "'b' " + op + " alpha")
+
+    # test against level 1
+    def test_equality_level1(self):
+        for op in ["==", "!="]:
+            run_test(self.series, "num " + op + " '3'")
+            run_test(self.series, "'3' " + op + " num")
+
+    def test_list_equality_level1(self):
+        for op in ["==", "!="]:
+            run_test(self.series, "num " + op + " ['3']")
+            run_test(self.series, "['3'] " + op + " num")
+
+    def test_in_operator(self):
+        for op in ["in", "not in"]:
+            run_test(self.series, "['3'] " + op + " num")
+            run_test(self.series, "'3' " + op + " num")
+
+
+class TestSeriesQueryByUnamedMultiIndex:
+    def setup_method(self, method):
+        self.series = Series(np.random.randn(10), index=[["a"] * 5 + ["b"] * 5, list(
+            range(10))])
+
+    def teardown_method(self, method):
+        del self.series
+
+    # check against first level
+    def test_query_first_level(self):
+        run_test(self.series, "ilevel_0 == 'b'")
+        run_test(self.series, "'b' == ilevel_0")
+
+    # check against not first level
+    def test_query_not_first_level(self):
+        run_test(self.series, "ilevel_1 > 4")
+        run_test(self.series, "4 > ilevel_1")
+
+    def test_both_levels(self):
+        for op in ["&", "|"]:
+            run_test(self.series, "(ilevel_0 == 'b') " + op + " ((ilevel_1 % 2) == 0)")
+            run_test(self.series, "((ilevel_1 % 2) == 0) " + op + " (ilevel_0 == 'b')")
```

Comment:
i wouldn't add any of this, rather i would take the dataframe tests and parameterimze them over Series and DataFrame.